File: algo/203. Remove Linked List Elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    var dummy = new ListNode(0);
    dummy.next = head;
    var cursor = dummy;
    while (cursor !== null && cursor.next !== null) {
        if (cursor.next.val == val) {
            cursor.next = cursor.next.next;
        } else {
            cursor = cursor.next;
        }
    }
    return dummy.next;
};

File: round2/203. Remove Linked List Elements.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode RemoveElements(ListNode head, int val) {
        if (head == null) return null;
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode pre = dummy, curr = head;
        while (curr != null) {
            if (curr.val != val) {
                pre = curr;
                curr = curr.next;
            } else {
                pre.next = curr.next;
                curr = curr.next;
            }
        }
        return dummy.next;
    }
}